File: 1. Labor/C#/Labor1_a_megoldas.cs
using System;

namespace Labor1_a_Solution
{
    class Program
    {
        static void Main(string[] args)
        {
            #region 1. FELADAT

            // Konzolra kiírás
            Console.Write("Hello World! \n");
            Console.WriteLine("Hello World Again!");

            #endregion

            #region 2. FELADAT

            // Kérjünk be valamit a felhasználótól és irjuk is ki azt
            Console.Write("Name: ");
            string name = Console.ReadLine();

            Console.WriteLine("Hello " + name);
            Console.WriteLine("Hello {0}", name);
            Console.WriteLine($"Hello '{name}'");

            #endregion

            #region 3. FELADAT

            // Kérjünk be egy számot, majd szorozzuk be egy a felhasználótól bekért számmal

            Console.Write("Number 1: ");
            int number1 = int.Parse(Console.ReadLine());

            Console.Write("Number 2: ");
            int number2 = int.Parse(Console.ReadLine());

            Console.WriteLine($"{number1 * number2}");
            #endregion

            #region 4. FELADAT

            // stringgé alakítása / Parse-olás / kasztolás

            // string
            string x = number1.ToString();

            // int
            string y = "4";
            int z = int.Parse(y);

            // kasztolas
            int valtozo = 255; // byte: 0-255
            byte masikValtozo = (byte)valtozo;

            #endregion

            #region 5. FELADAT

            /*
            Téglalap a és b oldalát bekérni, eltárolni egy-egy változóban
            majd a területét és kerületét kiszámolni és kiírni a következő formában:
            "Kerület={változó} [tabulátor] terület={változó}."
            */

            Console.WriteLine("Add meg az 'a' oldalt.");
            int aOldal = int.Parse(Console.ReadLine());
            Console.WriteLine("Add meg az 'b' oldalt.");
            int bOldal = int.Parse(Console.ReadLine());

            // kerület: (a+b)*2
            int kerulet = (aOldal + bOldal) * 2; // figyeljünk itt is a matematikailag helyes zárójelezésre!

            // terület: a*b
            int terulet = aOldal * bOldal;

            Console.WriteLine("Kerület={" + kerulet + "} \t Terület={" + terulet + "}");
            #endregion

            Console.ReadKey();
        }
    }
}

File: 1. Labor/TypeScript/Labor1_a_megoldas.ts
// #region 1. FELADAT
// Konzolra kiírás
console.log('Hello World!');
console.log('Hello World Again!');

// #endregion

// --------------------------

// #region 2. FELADAT
// Kérjünk be valamit a felhasználótól és irjuk is ki azt
const n: string = prompt('Name: ') || '';
console.log(`Hello ${n}`);

// #endregion

// --------------------------

// #region 3. FELADAT
// Kérjünk be egy számot, majd szorozzuk be egy a felhasználótól bekért számmal
const number1: number = parseInt(prompt('Number 1: ') || '0');
const number2: number = parseInt(prompt('Number 2: ') || '0');
console.log(number1 * number2);

// #endregion

// --------------------------

// #region 4. FELADAT
// stringgé alakítása / Parse-olás / kasztolás
// string
const x: string = number1.toString();

// int
const y: string = "4";
const z: number = parseInt(y);

// kasztolás
const valtozo: number = 255; // byte: 0-255
const masikValtozo: number = valtozo;

// #endregion

// --------------------------

// #region 5. FELADAT
// Téglalap a és b oldalát bekérni, eltárolni egy-egy változóban
// majd a területét és kerületét kiszámolni és kiírni a következő formában:
// "Kerület={változó} [tabulátor] terület={változó}."

const aOldal: number = parseInt(prompt("Add meg az 'a' oldalt: ") || '0');
const bOldal: number = parseInt(prompt("Add meg az 'b' oldalt: ") || '0');

// kerület: (a+b)*2
const kerulet: number = (aOldal + bOldal) * 2;

// terület: a*b
const terulet: number = aOldal * bOldal;

console.log(`Kerület=${kerulet} \t Terület=${terulet}`);

// #endregion
